Resolve issues with automorphisms and Graphs and more general things

lkastner opened the pull request:
- Remove module declaration `Graphs`, as it is only causing confusion, see #1563 and many others
- `automorphisms(Graph)` are now `automorphism_group_generators(Graph)` (the latter already exists for other things)
- Move tests of `Graph` to `Combinatorics` section

lkastner committed:
inner_automorphisms_group -> inner_automorphism_group

lkastner committed:
Remove module Graph

Apparently this was only causing confusion.

lkastner committed:
Graphs: Move tests of Graph to Combinatorics section.

lkastner committed:
Graph: automorphisms -> automorphism_group_generators

thofma commented:
OK, now it gets a bit more complicated. Since `automorphisms` was used in two different ways, we cannot use aliases to solve this (we can either make `automorphisms === automorphism_group_generators` or `automorphisms === automorphism_list` but not both).

So let's go down the deprecation route. I will push some changes here once I am done in Hecke.

thofma committed:
use deprecation instead of alias

thofma commented:
@lkastner Are there other automorphism group things you want to do here?

lkastner commented:
> @lkastner Are there other automorphism group things you want to do here?

I'll just add a paragraph to the style guide and fix the docs.

lkastner committed:
Graph: automorphisms is the right name in Polymake.jl

lkastner committed:
Styleguide: Add paragraph about function naming

lkastner committed:
Styleguide: Add explanation why we deviate from Julia

lkastner committed:
Styleguide: Wrap lines

fieker commented:
what about returning permutations or actual graph autmorphisms instead of Vec{Int}? Or should this be a seperate PR?

lkastner commented:
> what about returning permutations or actual graph autmorphisms instead of Vec{Int}? Or should this be a seperate PR?

Yes, I want to make a separate PR for these things. This is already large enough.

micjoswig commented:
Looks good to me.

fingolfin commented on docs/src/DeveloperDocumentation/styleguide.md:
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
+  functions do not have to follow these rules.) Types (and their constructor)
+  tend to be in `CamelCase`. However, please ALSO provide the constructor/ a
+  constructor in underscore_case. As a user I do usually not know if something
+  is a constructor or a function (nor do I want to).
```
```suggestion
  tend to be in `CamelCase`. However, please *also* provide the constructor (or a
  constructor) in `snake_case`. As a user one usually does not know if something
  is a constructor or a function.
```

fingolfin commented on docs/src/DeveloperDocumentation/styleguide.md:
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
+  constructor in underscore_case. As a user I do usually not know if something
+  is a constructor or a function (nor do I want to).
+- Noteworthy difference to Julia base is that we do not have exceptions `is*`
+  or `has*`.
```
```suggestion
- Noteworthy difference to Julia base is that we do not have exceptions for
  `is*` or `has*` as prefix.
```

fingolfin commented on docs/src/DeveloperDocumentation/styleguide.md:
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
+  or `has*`.
   It is `is_foo` instead of `isfoo` and `has_bar` instead of `hasbar`.
+  The main reason is to avoid awkward constructions like `isvery_ample`, while
+  also being consistent.
```
```suggestion
  also being consistent.
  For compatibility with standard Julia, while staying consistent internally,
  we also provide aliases (using `AbstractAlgebra.@alias`) for various standard
  Julia functions, e.g. `is_one` as alias for `isone`
```

fingolfin commented on docs/src/DeveloperDocumentation/styleguide.md:
```diff
@@ -13,31 +13,44 @@ apply to Oscar, too (that said, for various reasons our code still violates
+  also being consistent.
 - For generic concepts choose generic names, based on general algebraic
   concepts, preferably not special names from your area of speciality.
 - Use Julia conventions where applicable.
```
```suggestion
- Use Julia conventions where applicable and when they don't contradict our
  own rules above.
```

fingolfin commented:
@lkastner I was also editing the developer guidelines, and added some change suggestions here to make them match. Let me know if you prefer to merge this first, then I can also submit a separate PR with these changes later *shrug*

fingolfin commented on src/Deprecations.jl:
```diff
@@ -0,0 +1,3 @@
+# Deprecated in 0.10.*
+
+@deprecate automorphisms(x::Graph) automorphism_group_generators(x)
```
What about `inner_automorphisms_group`, should it also be added here?

lkastner replied:
There is an alias for this.

lkastner committed:
Apply suggestions from code review

Co-authored-by: Max Horn <max@quendi.de>

lkastner commented:
> @lkastner I was also editing the developer guidelines, and added some change suggestions here to make them match. Let me know if you prefer to merge this first, then I can also submit a separate PR with these changes later _shrug_

Thanks, I added your suggestions.

lkastner committed:
Update docs/src/DeveloperDocumentation/styleguide.md

Co-authored-by: Max Horn <max@quendi.de>
